src/lib/config-utils.ts

```
@@ -0,0 +1,376 @@
+}
+
+export function parseRpiBootConfig(confStr: string): ConfigOptions {
+	const conf: ConfigOptions = { };
+	const configStatements = confStr.split(/\r?\n/);
+
+	for (const configStr of configStatements) {
+		// Don't show warnings for comments and empty lines
+		const trimmed = _.trimStart(configStr);
+		if (_.startsWith(trimmed, '#') || trimmed === '') {
+			continue;
+		}
+		let keyValue = /^([^#=]+)=(.+)/.exec(configStr);
+		if (keyValue != null) {
+			if (!_.includes(rpiArrayConfigKeys, keyValue[1])) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				const key = keyValue[1];
+				if (conf[key] == null) {
+					conf[key] = [];
+				}
+				(conf[key] as string[]).push(keyValue[2]);
+			}
+		} else {
+			keyValue = /^(initramfs) (.+)/.exec(configStr);
+			if (keyValue != null) {
+				conf[keyValue[1]] = keyValue[2];
+			} else {
+				console.log(`Warning - Could not parse config.txt entry: ${configStr}. Ignoring.`);
+			}
+		}
+	}
+
+	return conf;
+}
+
+export function parseExtlinuxBootConfig(confStr: string): ConfigOptions {
+	const parsedBootFile = parseExtlinuxFile(confStr);
+
+	// First find the default label name
+	const defaultLabel = _.find(parsedBootFile.globals, (_v, l) => {
+		if (l === 'DEFAULT') {
+			return true;
+		}
+		return false;
+	});
+
+	if (defaultLabel == null) {
+		throw new Error('Could not find default entry for extlinux.conf file');
+	}
+
+	const labelEntry = parsedBootFile.labels[defaultLabel];
+
+	if (labelEntry == null) {
+		throw new Error(`Cannot find default label entry (label: ${defaultLabel}) for extlinux.conf file`);
+	}
+
+	// All configuration options come from the `APPEND` directive in the default label entry
+	const appendEntry = labelEntry.APPEND;
+
+	if (appendEntry == null) {
+		throw new Error('Could not find APPEND directive in default extlinux.conf boot entry');
+	}
+
+	const conf: ConfigOptions = { };
+	const values = appendEntry.split(' ');
+	for(const value of values) {
+		const parts = value.split('=');
```

Maybe raise an error if `parts.length isnt 2`?

---

Sure, will add.